### ISSUE TITLE
Improved FabIL Constructors

### ISSUE THREAD

**Issue by tmagrino** [apl-cornell/fabric#2]
Currently FabIL is a bit difficult to use due to some awkward conventions designed for the Fabric compiler.  In particular, the [constructor protocol for FabIL]( http://www.cs.cornell.edu/projects/fabric/manual/0.3.0/html/fabil.html#fabil-constructor-protocol)) is ugly and error prone.  We should find a way to make constructors intuitive in FabIL, so that they look more like conventional constructor calls.  Changing this:
```java
new MyObject@store().mypackage$MyObject$(args);
```
to this:
```java
new MyObject@store(args);
```

**Comment by K33TY** [comment]
Dr. Myers was mentioning improving Fabric constructors, as well.

**Comment by K33TY** [comment]
In the travel example, they are passing different arguments into both method calls, is there a reason for this?

Create.fil line 117:
`Account  airlineAccount = new Account@bankStore(bank).travel$Account$(airline);`

**Comment by tmagrino** [comment]
Yeah, that has to do with class type parameters, which FabIL does not support currently.
